python/rubin/rag/scrapers/scrape_discourse.py

```diff
@@ -0,0 +1,279 @@
+        return []
+    data = response.json()
+    return data.get("topic_list", {}).get("topics", [])
+
+
+class StatusCodeError(Exception):
+    """Custom exception class for bad requests response status code."""
+
+    def __init__(self, message: str) -> None:
+        self.message = message
+        super().__init__(self.message)
+
+    def __str__(self) -> str:
+        return self.message
+
+
+def get_posts_for_topic(topic_id: int) -> dict:
+    """
+    Get all posts for a Discourse topic.
+
+    Parameters
+    ----------
+    topic_id : int
+        The discourse topic identifier, which is an integer.
+
+    Returns
+    -------
+    dict
+        A dictionary containing the posts for the relevant
+        Discourse topic.
+    """
+    url = f"{DISCOURSE_URL}/t/{topic_id}.json"
+    response = requests.get(url, headers=headers, timeout=10)
+    if response.status_code != 200:
+        raise StatusCodeError(f"HTTP {response.status_code}")
+    data = response.json()
+    topic_title = data.get("title", f"topic_{topic_id}")
+    posts = data.get("post_stream", {}).get("posts", [])
+    cleaned_posts = []
+
+    for post in posts:
+        cleaned = clean_post(post)
+        cleaned_posts.append(cleaned)
+
+    return {
+        "topic_id": topic_id,
+        "topic_title": topic_title,
+        "posts": cleaned_posts,
+    }
+
+
+def clean_post(post: dict) -> dict:
+    """
+    Format the data in a Discourse post.
+
+    Parameters
+    ----------
+    post : dict
+        The post as retrieved from the Discourse API.
+
+    Returns
+    -------
+    dict
+        A well-formatted dictionary representing the post.
+    """
+    return {
+        "post_id": post.get("id"),
+        "username": post.get("username", "unknown"),
+        "created_at": post.get("created_at"),
+        "cooked": post.get("cooked", ""),
+        "raw": post.get("raw", ""),
+    }
+
+
+def topics_to_docs(topics: list) -> list:
+    """Convert Discourse topics to a list of LangChain docs.
+
+    Parameters
+    ----------
+    topics : list
+        list of Discourse topics, where each topic is a dictionary.
+
+    Returns
+    -------
+    docs : list
+       list of LangChain documents (not chunked).
+    """
+    docs = []
+    for topic in topics:
+        for post in topic["posts"]:
+            if post["cooked"] == "":
+                continue
+            metadata = post.copy()
+            del metadata["cooked"]
+            metadata["source_key"] = "discourse"
+            metadata["topic_id"] = topic["topic_id"]
+            metadata["topic_title"] = topic["topic_title"]
+            # https://community.lsst.org/t/10138
+            metadata["source"] = f"{DISCOURSE_URL}/t/{topic['topic_id']}"
+            doc = Document(page_content=post["cooked"], metadata=metadata)
+            docs.append(doc)
+
+    return docs
+
+
+def scrape_discourse(output_dir: str, *, max_pages: int | None = None) -> None:
```

Comment:
```suggestion
def scrape_discourse(yaml_path: str, output_dir: str, *, max_pages: int | None = None) -> None:
```
Add this so that it works with run_scraping.py. You can choose whether to use the yaml or not.

Reply:
Great, thanks. Done in https://github.com/lsst-dm/rubin_rag/pull/41/commits/48ecc6459ff3e88eaf18febe54f62992bc4ca4e5.